test/e2e/storage/non_graceful_node_shutdown.go

```diff
@@ -0,0 +1,184 @@
+				if dDriver, ok := driver.(storageframework.DynamicPVTestDriver); ok {
+					ginkgo.By("Creating a gce-pd storage class")
+					sc := dDriver.GetDynamicProvisionStorageClass(config, "")
+					_, err := c.StorageV1().StorageClasses().Create(context.TODO(), sc, metav1.CreateOptions{})
+					framework.ExpectNoError(err)
+					scName := &sc.Name
+
+					ginkgo.By("Creating a pvc using the gce-pd storage class")
+					pvc := e2epv.MakePersistentVolumeClaim(e2epv.PersistentVolumeClaimConfig{
+						StorageClassName: scName,
+					}, ns)
+					gotPVC, err := c.CoreV1().PersistentVolumeClaims(ns).Create(context.TODO(), pvc, metav1.CreateOptions{})
+					framework.ExpectNoError(err)
+
+					ginkgo.By("Creating a deployment using the pvc " + pvc.Name)
+					podLabels := map[string]string{"app": "sts-pod"}
+					dep := makeDeployment(ns, gotPVC.Name, podLabels)
+					_, err = c.AppsV1().Deployments(ns).Create(context.TODO(), dep, metav1.CreateOptions{})
+					framework.ExpectNoError(err)
+
+					ginkgo.By(fmt.Sprintf("Ensuring that the pod of deployment %s is running and ready", dep.Name))
+					labelSelector := labels.SelectorFromSet(labels.Set(podLabels))
+					podList, err := e2epod.WaitForPodsWithLabelRunningReady(c, ns, labelSelector, 1, podWaitTimeout)
+					framework.ExpectNoError(err)
+					pod := &podList.Items[0]
+					oldNodeName := pod.Spec.NodeName
+
+					ginkgo.By("Stopping the kubelet non gracefully for pod" + pod.Name)
+					utils.KubeletCommand(utils.KStop, c, pod)
+
+					ginkgo.By("Adding out of service taint on node " + oldNodeName)
+					// taint this node as out-of-service node
+					taint := v1.Taint{
+						Key:    v1.TaintNodeOutOfService,
+						Effect: v1.TaintEffectNoExecute,
+					}
+					e2enode.AddOrUpdateTaintOnNode(c, oldNodeName, taint)
+
+					ginkgo.By(fmt.Sprintf("Checking if the pod %s got rescheduled to a new node", pod.Name))
+					labelSelectorStr := labels.SelectorFromSet(podLabels).String()
+					podListOpts := metav1.ListOptions{
+						LabelSelector: labelSelectorStr,
+						FieldSelector: fields.OneTermNotEqualSelector("spec.nodeName", oldNodeName).String(),
+					}
+					_, err = e2epod.WaitForPodsWithListOptionsRunningReady(c, podListOpts, ns, 1)
+					framework.ExpectNoError(err)
+				} else {
+					framework.Logf("csi driver expected DynamicPVTestDriver but got %v", driver)
+				}
+			})
+		})
+	})
+})
+
+func makeDeployment(ns, pvcName string, labels map[string]string) *appsv1.Deployment {
+	ssReplicas := int32(1)
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "sts-pod",
+			Namespace: ns,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &ssReplicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  "sts-pod-nginx",
+							Image: e2epod.GetTestImage(imageutils.Nginx),
```

Comment:
Better using the agnhost image, I can't remember now which of the commands create the webserver, but there are several examples on the framework

Reply:
I have seen existing tests using that. Are we trying to get away with using this? I learnt that agnhost has some utilities which are helpful but I don't need any of those in this case.

Reply:
yeah, agnhost works for windows and linux, in addition, reducing the number of images that the e2e need to pull, improves the overall time to run

Reply:
Done. added